File: Lista Enlazada/Lista.cs
using System;

class Node {
    public int data;
    public Node next;
    
    public Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    static Node head;
    
    static void Main() {
        int choice = 0;
        while (choice != 9) {
            Console.WriteLine("\n*********Main Menu*********");
            Console.WriteLine("1. Agregar al inicio");
            Console.WriteLine("2. Agregar al final");
            Console.WriteLine("3. Agregar en una posicion aleatoria");
            Console.WriteLine("4. Eliminar desde el inicio");
            Console.WriteLine("5. Eliminar desde el final");
            Console.WriteLine("6. Eliminar desde una posicion aleatoria");
            Console.WriteLine("7. Mostrar");
            Console.WriteLine("8. Buscar");
            Console.WriteLine("9. Salir");
            Console.Write("Ingresa tu opciom: ");
            choice = Convert.ToInt32(Console.ReadLine());
            
            switch (choice) {
                case 1:
                    InsertBeg();
                    break;
                case 2:
                    InsertEnd();
                    break;
                case 3:
                    RandomInsert();
                    break;
                case 4:
                    DeleteBeg();
                    break;
                case 5:
                    DeleteEnd();
                    break;
                case 6:
                    RandomDelete();
                    break;
                case 7:
                    Display();
                    break;
                case 8:
                    Search();
                    break;
                case 9:
                    Environment.Exit(0);
                    break;
                default:
                    Console.WriteLine("Opcion invalida! Por favor intenta de nuevo.");
                    break;
            }
        }
    }
    
    static void InsertBeg() {
        Console.Write("Ingresa un elemento: ");
        int item = Convert.ToInt32(Console.ReadLine());
        Node newNode = new Node(item);
        newNode.next = head;
        head = newNode;
        Console.WriteLine("Elemento insertado");
    }
    
    static void InsertEnd() {
        Console.Write("Ingresa un elemento: ");
        int item = Convert.ToInt32(Console.ReadLine());
        Node newNode = new Node(item);
        
        if (head == null) {
            head = newNode;
            Console.WriteLine("Elemento insertado");
        } else {
            Node temp = head;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newNode;
            Console.WriteLine("Elemento insertado");
        }
    }
    
    static void RandomInsert() {
        Console.Write("Ingresa un elemento: ");
        int item = Convert.ToInt32(Console.ReadLine());
        Console.Write("Ingresa la posicion: ");
        int pos = Convert.ToInt32(Console.ReadLine());
        
        Node newNode = new Node(item);
        Node temp = head;
        
        for (int i = 0; i < pos - 1; i++) {
            if (temp == null) {
                Console.WriteLine("La posicion no existe");
                return;
            }
            temp = temp.next;
        }
        
        if (temp != null) {
            newNode.next = temp.next;
            temp.next = newNode;
            Console.WriteLine("Elemento insertado");
        } else {
            Console.WriteLine("La posicion no existe");
        }
    }
    
    static void DeleteBeg() {
        if (head == null) {
            Console.WriteLine("La lista esta vacia");
        } else {
            head = head.next;
            Console.WriteLine("Elemento eliminado desde el principio");
        }
    }
    
    static void DeleteEnd() {
        if (head == null) {
            Console.WriteLine("La lista esta vacia");
        } else if (head.next == null) {
            head = null;
            Console.WriteLine("Solo se elimino un nodo de la lista");
        } else {
            Node temp = head;
            Node prev = null;
            while (temp.next != null) {
                prev = temp;
                temp = temp.next;
            }
            prev.next = null;
            Console.WriteLine("Elemento eliminado desde el final");
        }
    }
    
    static void RandomDelete() {
        if (head == null) {
            Console.WriteLine("La lista esta vacia");
            return;
        }
        
        Console.Write("Ingresa la posicion: ");
        int pos = Convert.ToInt32(Console.ReadLine());
        
        if (pos == 0) {
            DeleteBeg();
            return;
        }
        
        Node temp = head;
        Node prev = null;
        
        for (int i = 0; i < pos; i++) {
            if (temp == null) {
                Console.WriteLine("La posicion no existe");
                return;
            }
            prev = temp;
            temp = temp.next;
        }
        
        if (temp != null) {
            prev.next = temp.next;
            Console.WriteLine("Elemento eliminado desde la posicion " + (pos + 1));
        } else {
            Console.WriteLine("La posicion no existe");
        }
    }
    
    static void Display() {
        if (head == null) {
            Console.WriteLine("La lista esta vacia");
        } else {
            Console.Write("Elementos en la lista: ");
            Node temp = head;
            while (temp != null) {
                Console.Write(temp.data + " ");
                temp = temp.next;
            }
            Console.WriteLine();
        }
    }
    
    static void Search() {
        if (head == null) {
            Console.WriteLine("La lista esta vacia");
            return;
        }
        
        Console.Write("Ingresa el elemento a buscar: ");
        int item = Convert.ToInt32(Console.ReadLine());
        
        Node temp = head;
        int pos = 0;
        bool found = false;
        
        while (temp != null) {
            if (temp.data == item) {
                Console.WriteLine("Elemento " + item + " encontrado en la posicion " + pos);
                found = true;
            }
            temp = temp.next;
            pos++;
        }
        
        if (!found) {
            Console.WriteLine("Elemento no encontrado en la lista");
        }
    }
}

File: Lista Enlazada/Lista.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

let head = null;

function insertbeg() {
    const item = parseInt(prompt("Ingresa un elemento: "));
    const newNode = new Node(item);
    newNode.next = head;
    head = newNode;
    console.log("Elemento insertado");
}

function insertend() {
    const item = parseInt(prompt("Ingresa un elemento: "));
    const newNode = new Node(item);
    
    if (head === null) {
        head = newNode;
        console.log("Elemento insertado");
    } else {
        let temp = head;
        while (temp.next !== null) {
            temp = temp.next;
        }
        temp.next = newNode;
        console.log("Elemento insertado");
    }
}

function randominsert() {
    const item = parseInt(prompt("Ingresa un elemento: "));
    const pos = parseInt(prompt("Ingresa la posicion: "));
    
    const newNode = new Node(item);
    let temp = head;
    
    for (let i = 0; i < pos - 1; i++) {
        if (temp === null) {
            console.log("La posicion no existe");
            return;
        }
        temp = temp.next;
    }
    
    if (temp !== null) {
        newNode.next = temp.next;
        temp.next = newNode;
        console.log("Elemento insertado");
    } else {
        console.log("La posicion no existe");
    }
}

function deletebeg() {
    if (head === null) {
        console.log("La lista esta vacia");
    } else {
        head = head.next;
        console.log("Elemento eliminado desde el principio");
    }
}

function deleteend() {
    if (head === null) {
        console.log("La lista esta vacia");
    } else if (head.next === null) {
        head = null;
        console.log("Solo se elimino un nodo de la lista");
    } else {
        let temp = head;
        let prev = null;
        while (temp.next !== null) {
            prev = temp;
            temp = temp.next;
        }
        prev.next = null;
        console.log("Elemento eliminado desde el final");
    }
}

function randomdelete() {
    if (head === null) {
        console.log("La lista esta vacia");
        return;
    }
    
    const pos = parseInt(prompt("Ingresa la posicion: "));
    
    if (pos === 0) {
        deletebeg();
        return;
    }
    
    let temp = head;
    let prev = null;
    
    for (let i = 0; i < pos; i++) {
        if (temp === null) {
            console.log("La posicion no existe");
            return;
        }
        prev = temp;
        temp = temp.next;
    }
    
    if (temp !== null) {
        prev.next = temp.next;
        console.log("Elemento eliminado desde la posicion " + (pos + 1));
    } else {
        console.log("La posicion no existe");
    }
}

function display() {
    if (head === null) {
        console.log("La lista esta vacia");
    } else {
        let temp = head;
        let result = "Elementos en la lista: ";
        while (temp !== null) {
            result += temp.data + " ";
            temp = temp.next;
        }
        console.log(result);
    }
}

function search() {
    if (head === null) {
        console.log("La lista esta vacia");
        return;
    }
    
    const item = parseInt(prompt("Ingresa el elemento a buscar: "));
    let temp = head;
    let pos = 0;
    let found = false;
    
    while (temp !== null) {
        if (temp.data === item) {
            console.log("Elemento " + item + " encontrado en la posicion " + pos);
            found = true;
        }
        temp = temp.next;
        pos++;
    }
    
    if (!found) {
        console.log("Elemento no encontrado en la lista");
    }
}

// Menú principal
function main() {
    let choice = 0;
    while (choice !== 9) {
        console.log("\n*********Main Menu*********");
        console.log("1. Agregar al inicio");
        console.log("2. Agregar al final");
        console.log("3. Agregar en una posicion aleatoria");
        console.log("4. Eliminar desde el inicio");
        console.log("5. Eliminar desde el final");
        console.log("6. Eliminar desde una posicion aleatoria");
        console.log("7. Mostrar");
        console.log("8. Buscar");
        console.log("9. Salir");
        choice = parseInt(prompt("Ingresa tu opcion: "));
        
        switch (choice) {
            case 1:
                insertbeg();
                break;
            case 2:
                insertend();
                break;
            case 3:
                randominsert();
                break;
            case 4:
                deletebeg();
                break;
            case 5:
                deleteend();
                break;
            case 6:
                randomdelete();
                break;
            case 7:
                display();
                break;
            case 8:
                search();
                break;
            case 9:
                return;
            default:
                console.log("Opcion invalida! Por favor intenta de nuevo.");
        }
    }
}


main();